### PR TITLE
Basic bcat:u implementation

### PR DESCRIPTION
**Pull request by shutterbug2000**
Enough of bcat to get Kirby and the Forgotten Land booting (aside from ReadCbufValue)

### CODE REVIEW

File: app/src/main/cpp/skyline/services/bcat/IBcatService.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#pragma once
+
+#include <services/serviceman.h>
+
+namespace skyline::service::bcat {
+    /**
+     * @brief IBcatService is used to interact with BCAT
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/services/bcat/IBcatService.h:10]: `used to interact with BCAT (Background Content Asymmetric synchronized delivery and Transmission)`

File: app/src/main/cpp/skyline/services/bcat/IServiceCreator.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        IServiceCreator(const DeviceState &state, ServiceManager &manager);
+
+        /**
+         * @brief Takes an input u64 ProcessId, returns an #IBcatService
+         * @url https://switchbrew.org/wiki/BCAT_services#bcat:a.2C_bcat:m.2C_bcat:u.2C_bcat:s
+         */
+        Result CreateBcatService(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response);
+
+        SERVICE_DECL(
+        SFUNC(0x0, IServiceCreator, CreateBcatService)
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/services/bcat/IServiceCreator.h:25]: Indentation is incorrect, run the auto-formatter on this file using `CTRL + ALT + L`.

File: app/src/main/cpp/skyline/services/bcat/IServiceCreator.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+#pragma once
+
+#include <services/serviceman.h>
+#include "IBcatService.h"
+
+namespace skyline::service::bcat {
+    /**
+     * @brief IApplicationProxyService is used to open an application proxy
+     * @url https://switchbrew.org/wiki/Applet_Manager_services#appletOE
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/services/bcat/IServiceCreator.h:12]: I assume you copy-pasted the comment here from AM, you should correct the URL and make the brief something along the lines of `IServiceCreator is used to create per-process/per-title service instances for BCAT`.

**Review by PixelyIon — CHANGES_REQUESTED** [review]
Good job, just a few nits here and there.

File: app/src/main/cpp/skyline/services/bcat/IBcatService.cpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/bcat/IBcatService.cpp:2]: Ngl I think it's 2022 but maybe you're right

File: app/src/main/cpp/skyline/services/bcat/IServiceCreator.cpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
```
Comment (bylaws) [inline app/src/main/cpp/skyline/services/bcat/IServiceCreator.cpp:2]: Same

**Review by PixelyIon — APPROVED** [review]
LGTM!

**Pull request merged**